src/odemis/driver/test/xt_client_test.py

```diff
@@ -563,41 +563,50 @@ def test_fwd_follows_z(self):
         # Set fwd follows z back to initial value
         self.microscope.set_fwd_follows_z(init_fwd_follows_z)
 
+    def test_pitch(self):
+        """Test setting and getting the pitch between two beams within the multiprobe pattern."""
+        pitch_range = self.microscope.pitch_info()["range"]
+        current_pitch = self.microscope.get_pitch()
+        new_pitch = current_pitch + 1
+        new_pitch = new_pitch if pitch_range[0] < new_pitch < pitch_range[1] else current_pitch - 1
+        self.microscope.set_pitch(new_pitch)
+        self.assertAlmostEqual(new_pitch, self.microscope.get_pitch())
+        self.microscope.set_pitch(current_pitch)
+
     def test_stigmator(self):
         """Test setting and getting stigmator values."""
         stig_range = self.microscope.stigmator_info()["range"]
         init_stig = self.microscope.get_stigmator()
+        self.assertIsInstance(init_stig, tuple)
         stig_x, stig_y = (init_stig[0] + 1e-3, init_stig[1] + 1e-3)
-        stig_x = stig_x if stig_range['x'][0] < stig_x < stig_range['x'][1] else init_stig[0] - 1e-6
-        stig_y = stig_y if stig_range['y'][0] < stig_y < stig_range['y'][1] else init_stig[1] - 1e-6
+        stig_x = stig_x if stig_range['x'][0] < stig_x <= stig_range['x'][1] else init_stig[0] - 1e-6
+        stig_y = stig_y if stig_range['y'][0] < stig_y <= stig_range['y'][1] else init_stig[1] - 1e-6
         new_stig = (stig_x, stig_y)
         self.microscope.set_stigmator(*new_stig)
         self.assertEqual(self.microscope.get_stigmator()[0], new_stig[0])
         self.assertEqual(self.microscope.get_stigmator()[1], new_stig[1])
+        # Try to set value outside of range
+        with self.assertRaises(OSError):
+            self.microscope.set_stigmator(stig_range['x'][1] + 1e-3, stig_range['y'][1] + 1e-3)
+        # Set back to the initial stigmator value, so the system is not misaligned after finishing the test.
```

Comment:
I think it is nice to check after this call (which raise an error) if the values are still the same as before.
This would catch if somehow and incorrect value might reset something to another value or do something else which is weird.

So here that would be repeading these two lines from above:

> self.assertEqual(self.microscope.get_stigmator()[0], new_stig[0])
> self.assertEqual(self.microscope.get_stigmator()[1], new_stig[1])